compiler/rustc_mir_transform/src/inline.rs

```diff
@@ -375,7 +375,7 @@ impl<'tcx> Inliner<'tcx> {
             return Err("incompatible sanitizer set");
         }
 
-        if callee_attrs.instruction_set != self.codegen_fn_attrs.instruction_set {
+        if callee_attrs.instruction_set.is_some() && callee_attrs.instruction_set != self.codegen_fn_attrs.instruction_set {
```

Comment:
Leave a comment explaining why having no instruction set allows you to get inlined into a function with such a flag